modules/messages/src/lib.rs

```diff
@@ -314,24 +316,41 @@ pub mod pallet {
 					}
 				}
 
+				let mut lane_messages_received_status =
+					Vec::with_capacity(lane_data.messages.len());
+				let mut is_lane_processing_stopped_no_weight_left = false;
+
 				for mut message in lane_data.messages {
 					debug_assert_eq!(message.key.lane_id, lane_id);
+					total_messages += 1;
+
+					if is_lane_processing_stopped_no_weight_left {
+						lane_messages_received_status.push((
+							message.key.nonce,
+							NotDispatchedReason::NotEnoughWeightForLane.into(),
```

Comment:
Probably we can just leave single `NotDispatchedReason::NotEnoughWeightForLane` instead of duplicating it for every nonce? I mean - all remaining messages will be rejected, so probably single `NotEnoughWeightForLane` would be enough? I admit that the current version is better for understanding though :)

Reply:
I was not 100% sure about this part, but I wanted to have all messages from request in the result :)
so I removed value `NotEnoughWeightForLane` and collect all skipped messages to the `skipped_for_not_enough_weight: Vec<MessageNonce>`  so there will be everything in the result without duplicating anything

Reply:
Cool, thank you! :)